### PR TITLE
CR-1141309 Removed the conditions done for sw_emu to return the memid…

### PR DESCRIPTION
**Pull request by venkatp-xilinx**
CR-1141309 Removed the conditions done for sw_emu to return the memidx as 0 as we have updated the RTD of sw_emu to be similar to hw_emu.

This is DSV and Libraries team escalated CR. This addressed ~4 CRs. Please review it on high priority.

#### Problem solved by the commit
Updated the RTD file in sw_emu to be similar to the hw_emu in all the scenarios including the HBM bus notation. So memidx should return based on the RTD file, not 0.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
None of the PRs introduced this issue. We got this issue when we addressed the HBM memory support.

#### Risks (if any) associated the changes in the commit
No Risk

#### What has been tested and how, request additional testing if necessary
Ran Various designs involving different memory configurations. And Ran the Vitis Canaries to ensure none of the existing tests are impacted

#### Documentation impact (if any)
No

### CODE REVIEW

**Review by chvamshi-xilinx — APPROVED** [review]
Looks good and much needed cleanup.
Only concern is about the timing of this change. Please do monitor regressions after this change is merged.

**Comment by venkatp36** [comment]
> Looks good and much needed cleanup. Only concern is about the timing of this change. Please do monitor regressions after this change is merged.

Thanks for the review Vamshi. I have done verification of various memory configurations and canary is also clean with this change.

**Comment by venkatp36** [comment]
Please do not merge this PR. We would like to handle it for 2023.1 release. I am unable to close the PR. @chvamshi-xilinx, please close the PR.

**Comment by chvamshi-xilinx** [comment]
Closed based on the request

**Pull request closed without merging**
